### ISSUE TITLE
Resolution change does nothing (LWJGL)

### ISSUE THREAD

**Issue by Crushy** [nifty-gui/nifty-gui#122]
I can't get nifty.resolutionChanged() to work on my PC (Nvidia GTX 460). The resolution changer on this  [demo](http://nifty-gui.sourceforge.net/webstart/nifty-default-controls-examples-1.3.jnlp) also doesn't seem to work.

This is what happens if I press restart screen without even changing the resolution.
![resize](https://f.cloud.github.com/assets/4438626/523115/45daea68-c097-11e2-84dd-1855e1d9a4e3.png)

This is what happens in a simple test case of the BatchRenderDevice snapshot, using Display.setResizable(true); and Display.wasResized() calling nifty.resolutionChanged();
![example](https://f.cloud.github.com/assets/4438626/523128/6d76823e-c098-11e2-994d-b777689bb5fe.png)

**Comment by void256** [comment]
I don't know whats the problem with the standard controls demo on your system. It works for me and I can "restart" the screen as well as change the resolutions with the combobox o_O

What's currently not yet supported is the standard controls demo when running it with the batched renderer. But for this you'll need to build it from source. So the jnlp should be fine.

nifty.resolutionChanged() is used to notify Nifty that the LWJGL display resolution was changed external to Nifty. Nifty will then use the current dimensions of the Display and relayout everything according to the new size.

What's wrong with your second screenshot? If you resize your window to that size then Nifty will adjust it and IMHO you'll get exactly what is visible in your second screenshot.

**Comment by Crushy** [comment]
I was expecting the text not to turn out compressed. If I start the program out as 600x600 or 800x600 the text looks fine. However, if I resize it to one of those resolutions, it starts being scaled. I just want to make sure if this is intentional.

As for the controls demo, I have no idea what's going on either. I can try tearing the code apart when I have the time. The upper buttons work but the mouse has a strange offset. The lower buttons and the combobox seem to be completely inoperative after a resize tough.

**Comment by Crushy** [comment]
I found myself with some time to spare and ran the demo on the following machines, running Windows 7-8 64 bit. 
Nvidia GTX450
Nvidia GT 330M
Radeon HD 4500

They all seem to share the same problem with resizing on the control demo, same as the screenshot.

**Comment by void256** [comment]
Did you try it with a very latest nightly build of 1.3.3-SNAPSHOT? And if not can you do so?

There have been some issues in the batched renderer that I've fixed just a couple of days ago. See this commit for details: 392171c5e2becc83b3bee904282cd9c33dd3169f

Did anything change with the very latest 1.3.3 Nifty? (Please note that these changes have not been merged to the 1.4 branch ...)

**Comment by Crushy** [comment]
I'm not sure if I was clear but the results I just mentioned were just running the controllers demo as-is.

I can confirm however that "nifty-lwjgl-renderer-1.3.3-20130608.223847-28" fixes all resizing issues on my current project.

**Comment by void256** [comment]
Thanks for the clarification :)

So the only remaining problem in this issue is the weird acting of the 1.3 standard controls demo when started as-is from the webstart URL, right? I can only try it on Windows 7 the next time I boot into it but I still don't have an explanation for that odd behaviour really :(

**Comment by Crushy** [comment]
Yeah, I did some quick checking on the windows systems and it seems everything works perfectly with the new renderer.

I tested the controls demo on my netbook, running Linux, with an Intel onboard GPU, as well as a friend of mine's Windows running a geforce 9600GT  and they all seemed to have the same issues.

It seems like there's something very specific going wrong here... I'm going to try resizing using the old 1.3 renderer tomorrow and see if it's only effecting that demo.

**Comment by Hammerkeg** [comment]
Quite sorry for the very late comment, but I have run into the same problem, have tried the latest 1.3.3-SNAPSHOT, that did not however fix the issue. The default controls example doesn't work either. Any idea what could be causing the problem? Could it be an issue with the graphics adapter? Is there any way to get the computer to render OpenGL through the CPU as a test?
